### DISCUSSION TITLE
What Wallet Connect product should I use for Angular?

### DISCUSSION THREAD

**Discussion by sebastiancordea97** [WalletConnect/walletconnect-monorepo#3348] [Developer Support]
Hello. Till now we had metamask injected in the browser to connect to an Angular app and sign some messages. I tried to integrate web3Modal the HTML one but i can not get the listeners to tell me when the account has changed or the chain has changed. In the metamask i used something like: 

public listenForChainChanges(): Observable<string> {
    this.provider?.on("chainChanged", (chainId: string) => {
      this.chainChangeSubject.next(chainId);
    });

    return this.chainChangeSubject.asObservable();
  }
  
  But for wallet connect i cannot get the onChainChanged listener from the  
  this.web3Modal.subscribeModal(async (newState: any) => {
      this.provider = newState;      

Also: 
 - Is web3Modal stable to use with other framework other than React? Because i have a lot of issues with it
 - is this guide up to date? https://docs.cloud.coinbase.com/wallet-sdk/docs/web3modal
 - why is standalone version not available anymore in the docs?
 - will there be an Angular example in the future?
 
 Thank you!

**Comment by glitch-txs** [comment]
There's a current issue with [chain switching and MetaMask](https://github.com/MetaMask/metamask-mobile/issues/6655)

**Reply by glitch-txs** [reply]
also, this was build from a dev from the community, might be helpful as well [Angular example](https://github.com/csurbier/walletconnectv2-angular)

**Reply by glitch-txs** [reply]
One last comment, you can use wagmi actions to track accounts/chains check https://wagmi.sh/core/actions/watchNetwork
